types/src/credentials/keys.rs

```diff
@@ -32,38 +34,44 @@ use crate::crypto::{
         DEFAULT_SIGNATURE_SCHEME,
         private_keys::{SigningKey, VerifyingKey},
         signable::Signable,
-        traits::{SigningKeyBehaviour, VerifyingKeyBehaviour},
     },
 };
 
 use thiserror::Error;
 
 use super::ClientCredential;
 
+#[derive(
+    Clone, Debug, TlsSerialize, TlsDeserializeBytes, TlsSize, Eq, PartialEq, Serialize, Deserialize,
```

Comment:
Only `Clone`, `Debug`, `Serialize` and `Deserialize` is needed.

Maybe it makes sense to do manual impls of Close, Serde, TlsSerde and Eq on the tagged type. Then the tags have to impl only Debug and nothing else.